thrust/thrust/detail/pointer.h

```diff
@@ -51,59 +51,47 @@ THRUST_NAMESPACE_END
 // Specialize `std::iterator_traits` (picked up by cuda::std::iterator_traits) to avoid problems with the name of
 // pointer's constructor shadowing its nested pointer type. We do this before pointer is defined so the specialization
 // is correctly used inside the definition.
-namespace std
-{
 template <typename Element, typename Tag, typename Reference, typename Derived>
-struct iterator_traits<THRUST_NS_QUALIFIER::pointer<Element, Tag, Reference, Derived>>
+struct std::iterator_traits<THRUST_NS_QUALIFIER::pointer<Element, Tag, Reference, Derived>>
```

Comment:
Should we specialize `_CUDA_VSTD::iterator_traits`

Reply:
Wouldn't this be a breaking change? What about users relying on the `std::iterator_traits` specializations?

Reply:
Yeah, but we also need to work around this for NVRTC when the time comes

Reply:
Let's revisit this when the time comes